SensusService/Protocol.cs

```diff
@@ -300,14 +301,14 @@ public static void DisplayAndStartAsync(Protocol protocol)
                         else
                         {
                             protocolsPage = new ProtocolsPage();
-                            await Application.Current.MainPage.Navigation.PushAsync(protocolsPage);
+                            SensusServiceHelper.Get().MainThreadSynchronizer.ExecuteThreadSafe(protocolsPage.Bind);
```

Comment:
Sorry, Wes's stuff. I didn't realize some o f his changes had been mixed in with mine.